### PR TITLE
fix(pypi): use merge strategy to get latest package versions

### LINKED ISSUE #6397: [python] Support for --extra-index-url
**What Renovate type are you using?**
We are using latest version of renovate bot, self-hosted integrated with GitLab. 

**Describe the bug**
I have found some issue that should be already resolved wrt support for --extra-index-url, but we are not able to set it up. I am not able to find anything related to it in documentation, so all our attempts were based on code. 

**Did you see anything helpful in debug logs?**
In logs it shows that repo was processed but no change/upgrade was detected. 

**To Reproduce**
We got the requirements.txt file looks like this 

```
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --allow-unsafe --output-file=./py3-requirements.txt ./requirements.in
#
--extra-index-url https://our_private.pypi/pypi
some-private-package==1.1.45 //really old version
```

We have also setup the host rules like this:
```
[
  {
    "domainName": "private.pypi.domain", // tried all the options here, eg hostType, domainName...
    "username": "our_awesome_renovatebot_user", //user to private pypi
    "password": "our_awesome_renovatebot_user_password",
    "hostType": "pypi"
  }
]
```

**Additional context**
According to for example https://github.com/renovatebot/renovate/issues/4035 it should be solved already, so maybe we are just missing something?

### PR DESCRIPTION
**Pull request by aexvir**
👋 Hi there,

This PR should fix the issue reported in https://github.com/renovatebot/renovate/issues/6397, where renovatebot wouldn't return the latest package version available on any of the scanned indexes.

With this change, renovatebot will behave like `pip` does.

I've also added tests for checking this behaviour 🙂 

closes  #6397

### CODE REVIEW

File: lib/datasource/pypi/index.spec.ts [hunk 1]
```diff
@@ -92,18 +96,28 @@ describe('datasource/pypi', () => {
...
         ...config,
         datasource,
         depName: 'azure-cli-monitor',
       });
+      expect(
+        res.releases.reduce((x, y) => (semver.gt(x.version, y.version) ? x : y))
+      ).toMatchObject({
+        version: '0.2.15',
+        releaseTimestamp: '2019-06-18T13:58:55',
+      });
```
Comment (viceice) [inline lib/datasource/pypi/index.spec.ts:120]: You don't need to reduce, the `res.releases` should be ordered ascending, so `res.releases.pop()` should be latest version.

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/datasource/pypi/index.spec.ts (hunk 1 above)
Comment (aexvir) [reply lib/datasource/pypi/index.spec.ts:120]: Alright, I wasn't sure if it was guaranteed to be ordered and I didn't had much time to dig deep into it 🙂 
Thanks!

Comment (aexvir) [reply lib/datasource/pypi/index.spec.ts]: Changed 🙂

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
@aexvir Will this fix  #6397 ?

**Comment by aexvir** [comment]
Yep. When we reported the issue we believed the cause was a different one, but as @rarkins pointed out, the issue was on the behaviour of returning on the first result, instead of something wrong on the logic for supporting multiple registries.

**Pull request merged**